src/frontend/src/optimizer/plan_expr_visitor/strong.rs

```diff
@@ -0,0 +1,340 @@
+    #[allow(dead_code)]
+    pub fn is_null(expr: &ExprImpl, null_columns: FixedBitSet) -> bool {
+        let strong = Strong::new(null_columns);
+        strong.is_null_visit(expr)
+    }
+
+    fn is_input_ref_null(&self, input_ref: &InputRef) -> bool {
+        self.null_columns.contains(input_ref.index())
+    }
+
+    fn is_null_visit(&self, expr: &ExprImpl) -> bool {
+        match expr {
+            ExprImpl::InputRef(input_ref) => self.is_input_ref_null(input_ref),
+            ExprImpl::Literal(literal) => literal.get_data().is_none(),
+            ExprImpl::FunctionCall(func_call) => self.is_null_function_call(func_call),
+            ExprImpl::FunctionCallWithLambda(_) => false,
+            ExprImpl::AggCall(_) => false,
+            ExprImpl::Subquery(_) => false,
+            ExprImpl::CorrelatedInputRef(_) => false,
+            ExprImpl::TableFunction(_) => false,
+            ExprImpl::WindowFunction(_) => false,
+            ExprImpl::UserDefinedFunction(_) => false,
+            ExprImpl::Parameter(_) => false,
+            ExprImpl::Now(_) => false,
+        }
+    }
+
+    fn is_null_function_call(&self, func_call: &FunctionCall) -> bool {
+        match func_call.func_type() {
+            // NOT NULL: This kind of expression is never null. No need to look at its arguments, if it has any.
+            ExprType::IsNull
+            | ExprType::IsNotNull
+            | ExprType::IsDistinctFrom
+            | ExprType::IsNotDistinctFrom
+            | ExprType::IsTrue
+            | ExprType::IsNotTrue
+            | ExprType::IsFalse
+            | ExprType::IsNotFalse => false,
+            // ANY: This kind of expression is null if and only if at least one of its arguments is null.
+            ExprType::Not
+            | ExprType::Equal
+            | ExprType::NotEqual
+            | ExprType::LessThan
+            | ExprType::LessThanOrEqual
+            | ExprType::GreaterThan
+            | ExprType::GreaterThanOrEqual
+            | ExprType::Like
+            | ExprType::Add
+            | ExprType::AddWithTimeZone
+            | ExprType::Subtract
+            | ExprType::Multiply
+            | ExprType::Modulus
+            | ExprType::Divide
+            | ExprType::Cast
+            | ExprType::Trim
+            | ExprType::Ltrim
+            | ExprType::Rtrim
+            | ExprType::Ceil
+            | ExprType::Floor
+            | ExprType::Extract
+            | ExprType::Greatest
+            | ExprType::Least => self.any_null(func_call),
+            // ALL: This kind of expression is null if and only if all of its arguments are null.
+            ExprType::And | ExprType::Or | ExprType::Coalesce => self.all_null(func_call),
+            // TODO: Function like case when is important but current its structure is complicated, so we need to implement it later if necessary.
+            // Assume that any other expressions cannot be simplified.
+            _ => false,
```

Comment:
Can we avoid using `_` here? Explicitly listing all the remaining variants will remind developers to update the code here for introducing new expressions in the future.

Reply:
Sure. There are about 200+ remaining exprs 🥵. Actually, I just analyze the most common exprs.